File: coding/train_yandex/2024/fast_track/task_C.py
import sys


def main():
    n = int(sys.stdin.readline())
    temps = list(map(int, sys.stdin.readline().split()))
    b = int(sys.stdin.readline())
    ans = []
    for _ in range(b):
        t_min, t_max, d_min, d_max = list(map(int, sys.stdin.readline().split()))
        left = n + d_min
        right = n + d_max
        min_temp = 10 ** 9 + 1
        max_temp = (-1) * min_temp
        for temp in temps[left:right + 1]:
            min_temp = min(min_temp, temp)
            max_temp = max(max_temp, temp)
        low = t_min > max_temp
        high = min_temp > t_max
        if low or high:
            ans.append("no")
            # sys.stdout.write("no\n")
        else:
            ans.append("yes")
            # sys.stdout.write("yes\n")

    sys.stdout.write("\n".join(ans))

if __name__ == '__main__':
    main()

File: coding/train_yandex/2024/fast_track/task_C.cpp
//https://contest.yandex.ru/contest/59886/problems/C/
// Created by omega515 on 14.03.24.
//

#include<bits/stdc++.h>

using namespace std;

const long long MAX_TEMP_CONST = pow(10, 9) + 1;

int main() {
    ios::sync_with_stdio(false);
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    long long n;
    cin >> n;
    vector<long long> temps(2 * n + 1);
    for (long long i = 0; i < 2 * n + 1; ++i) {
        cin >> temps[i];
    }
    long long b, tMin, tMax, dMin, dMax, left, right;
    cin >> b;
    for (long long k = 0; k < b; ++k) {
        cin >> tMin >> tMax >> dMin >> dMax;
        left = n + dMin;
        right = n + dMax;
        long long min_temp = MAX_TEMP_CONST;
        long long max_temp = -MAX_TEMP_CONST;
        for (long long i = left; i < right + 1; ++i) {
            long long temp = temps[i];
            min_temp = min(min_temp, temp);
            max_temp = max(max_temp, temp);
        }
        bool low = tMin > max_temp;
        bool high = min_temp > tMax;
        if (low || high) {
            cout << "no\n";
        } else {
            cout << "yes\n";
        }

    }


    return 0;
}
